amplify/backend/function/dogfunction/src/createDog.js

```diff
@@ -0,0 +1,40 @@
+/* Amplify Params - DO NOT EDIT
+You can access the following resource attributes as environment variables from your Lambda function
+var environment = process.env.ENV
+var region = process.env.REGION
+var storageDogtableName = process.env.STORAGE_DOGTABLE_NAME
+var storageDogtableArn = process.env.STORAGE_DOGTABLE_ARN
+
+Amplify Params - DO NOT EDIT */
+
+var AWS = require("aws-sdk");
+var region = process.env.REGION;
+AWS.config.update({ region: region });
+var documentClient = new AWS.DynamoDB.DocumentClient({ region: region });
+var ddb_table_name = process.env.STORAGE_DOGTABLE_NAME
+
+function write(params, event, callback) {
+  documentClient.put(params, function(err, data) {
+    if (err) {
+      callback(err);
```

Comment:
shouldn't we throw an error instead of pass the `error` to the callback?

Reply:
maybe console.log out the error as well?

Reply:
We pass the error to the callback, which is the front-end graphql operation function. So that we could access the error in the front-end. Otherwise you could only be able to check the error through cloudwatch